restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/fcm/FcmMessagingService.java

```diff
@@ -42,13 +42,19 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
 
     private void sendNotification() {
         //start service
-        Intent intent = new Intent(this, RCDevice.class);
-        intent.setAction(RCDevice.ACTION_FCM);
-        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O){
-            startService(intent);
-        } else{
-            startService(intent);
+        //we dont need to start a service if its already running.
+        //The restcomm server will always send the push...
+        if (!RCDevice.isServiceAttached){
```

Comment:
Isn't there a chance that RCDevice is null at this point? Should we guard for that too?

Reply:
property is static; I think we are good.